### PR TITLE
Initial commit for API project

### PR DESCRIPTION
**Pull request by wr46**
- Initial project base structure (NestJs)
- Docker integration for the database and adminer (postgres)
- Project documentation

### CODE REVIEW

File: api/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+https://www.contributor-covenant.org/version/2/0/code_of_conduct.html.
+
+Community Impact Guidelines were inspired by [Mozilla's code of conduct
+enforcement ladder](https://github.com/mozilla/diversity).
+
+[homepage]: https://www.contributor-covenant.org
+
+For answers to common questions about this code of conduct, see the FAQ at
+https://www.contributor-covenant.org/faq. Translations are available at
+https://www.contributor-covenant.org/translations.
```
Comment (comoser) [inline api/CODE_OF_CONDUCT.md]: No need for a CoC specific for the API, it's already in the `root`

File: api/package.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
+{
+    "name": "api",
+    "version": "0.0.1",
+    "description": "",
+    "author": "",
+    "private": true,
+    "license": "UNLICENSED",
```
Comment (comoser) [inline api/package.json]: MIT is the base license

**Comment by wr46** [comment]
> I wouldn’t forget to add phone number to the user records, before we all submit the same PR

This User entity work as an exemple for project structure.

File: api/src/users/users.service.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { Injectable, Inject } from "@nestjs/common";
+import { Repository } from "typeorm/repository/Repository";
+import { User } from "./entities/user.entity";
+import { CreateUserDto } from "./dto/create-user.dto";
+import { USER_REPOSITORY } from "src/constants";
+
+@Injectable()
+export class UsersService {
```
Comment (comoser) [inline api/src/users/users.service.ts]: suggestion here: you can actually create a generic `repo` which you called `service` and extend from it with the Type inside the <>, like <User>, just to enforce the usage of at least standard methods, like `find`, `findAll`, `findOne`, `create` and `remove`

**Comment by comoser** [comment]
@all-contributors please add @wr46  for code, ideas, "project management", maintenance, documentation

File: api/src/users/users.service.ts (hunk 3 above)
Comment (wr46) [reply api/src/users/users.service.ts]: This repository must be changed to a simpler structure.

**Comment by sabind** [comment]
#107 adds some missing pieces to the heroku build pipeline.

File: api/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+module.exports = {
```
Comment (s0) [inline api/.eslintrc.js]: This should extend the base typescript eslint ruleset `../.eslintrc-ts.js` like this: https://github.com/reach4help/reach4help/blob/master/map/.eslintrc.js

File: api/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    "module": "commonjs",
+    "declaration": true,
+    "removeComments": true,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "target": "es2017",
+    "sourceMap": true,
+    "outDir": "./dist",
+    "baseUrl": "./",
+    "incremental": true
```
Comment (s0) [inline api/tsconfig.json]: Running `start`, `build` or `start:dev` are all giving me typescript errors atm, looks like duplicate definitions for standard libraries.

I think the issue here comes from `styled-components` being installed in some of the other packages in the repo, and polluting the shared `node_modules`, but we'll need this here to be able to compile the api now.

cc @comoser 

Adding `skipLibCheck` seems to fix the issue.

```suggestion
    "incremental": true,
    "skipLibCheck": true
```

**Comment by s0** [comment]
I'm reviewing this atm

File: api/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+```
+yarn typeorm migration:run
+```
+
+6 - The api swagger-ui is now available under [localhost:3001](http://localhost:3001)
+
+
+# How to report a bug
+
+> If you find a security vulnerability, open an issue describing the problem.
```
Comment (s0) [inline api/CONTRIBUTING.md]: We don't want people opening issues to report security vulnerabilities, we want them to disclose ethically. @mdeous will be writing up and publishing our security policy soon, this readme should refer to that when available, until then, change this to the following:

```suggestion
> If you find a security vulnerability, please send a report to <security@reach4help.org>
```

File: api/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+### Code Style Enforced by Prettier
+
+Prettier guarantees the code style adopted and runs on commit, stick to this code style.
+
+<img height="42" width="42" src="https://prettier.io/icon.png" alt="Prettier">[ Prettier site for more information](https://prettier.io/ 'Prettier site for more information')
+
+# PULL Requests
+
+Opening pull requests should be done with enough information and screenshots for visual changes to facilitate the reviewers job. Its MANDATORY to add a link to the issue related
```
Comment (s0) [inline api/CONTRIBUTING.md]: Visual changes are not relevant for `api`.

File: api/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
+# Contributing to web-client for Reach4Help
```
Comment (s0) [inline api/CONTRIBUTING.md]: It's `api` not `web-client` :)

File: api/.eslintrc.js (hunk 4 above)
Comment (wr46) [reply api/.eslintrc.js]: We have to make some changes. I did it but then many strange errors appeared.

File: api/tsconfig.json (hunk 5 above)
Comment (comoser) [reply api/tsconfig.json]: We should be able to specify the `styled-components` dependency only on the web-client, is there a way @s0 ?

Comment (s0) [reply api/tsconfig.json]: With `yarn workspaces`, I don't believe that's possible, unless another package in the workspace is using a different version of that package.

I've used [rush](https://rushjs.io/) in other repos alongside pnpm, which has considerably more advanced dependency management, and prevents phantom dependencies. I prefer it and typically use it on my own projects, the downside however is it's a lot more opinionated, and a little bit more fiddly to add / remove packages, and has a bit more of a learning curve compared to yarn workspaces.

Comment (s0) [reply api/tsconfig.json]: And of course it's yet another development tool to install...

Comment (s0) [reply api/tsconfig.json]: For now, i'd recommend just adding the skip lib check in the projects that need it, and continuing with yarn workspaces.

File: api/.eslintrc.js (hunk 4 above)
Comment (s0) [reply api/.eslintrc.js]: what were the errors?

**Comment by s0** [comment]
I'm happy for it to be merged if @comoser is :)

**Pull request merged**